smb/src/main/java/ch/cyberduck/core/smb/SMBListService.java

```diff
@@ -40,9 +41,16 @@ public SMBListService(SMBSession session) {
     @Override
     public AttributedList<Path> list(final Path directory, final ListProgressListener listener) throws BackgroundException {
 
+        String dst = directory.getAbsolute();
+        if(dst.charAt(0) == '/') {
```

Comment:
Should we have generic handling of this path conversion for all features that work with absolute paths from `Path`?

Reply:
The other calls seem to work with both formats.
But to make it the same everywhere, we have now added the class [SMBUtils](https://github.com/mbscit/smb-cyberduck/blob/feature/final-clean/smb/src/main/java/ch/cyberduck/core/smb/SMBUtils.java) that does this conversion, and applied it everywhere the absolute path is used.